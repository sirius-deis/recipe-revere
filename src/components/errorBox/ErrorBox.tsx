import { FC } from "react";
import styles from "./ErrorBox.module.css";

interface ErrorBoxProps {
  message: string;
}

const ErrorBox: FC<ErrorBoxProps> = ({ message }) => {
  return <div className={styles.error}>{message}</div>;
};

export default ErrorBox;
