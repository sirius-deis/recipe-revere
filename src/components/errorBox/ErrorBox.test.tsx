import { render, screen } from "@testing-library/react";
import ErrorBox from "./ErrorBox";

describe("ErrorBox component", () => {
  it("should match snapshot", () => {
    const { container } = render(<ErrorBox message="Error message" />);
    expect(container).toMatchSnapshot();
  });
  it("should render component with proper text", () => {
    render(<ErrorBox message="Error message" />);
    expect(screen.getByText("Error message")).toBeInTheDocument();
  });
});
