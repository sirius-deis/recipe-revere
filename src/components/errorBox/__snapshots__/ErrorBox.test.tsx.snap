// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBox component should match snapshot 1`] = `
<div>
  <div
    class="error"
  >
    Error message
  </div>
</div>
`;
